File: clojure/secret-handshake/src/secret_handshake.clj
(ns secret-handshake)

(defn commands [handshake]
  (cond-> []
          (bit-test handshake 0) (conj "wink")
          (bit-test handshake 1) (conj "double blink")
          (bit-test handshake 2) (conj "close your eyes")
          (bit-test handshake 3) (conj "jump")
          (bit-test handshake 4) reverse))

File: ruby/secret-handshake/secret_handshake.rb
class Array
  def reverse_if(flag)
    flag ? self.reverse : self
  end
end

class SecretHandshake
  EVENTS = ["wink", "double blink", "close your eyes", "jump"]
  REVERSE_EVENT = 1 << 4

  def initialize(handshake)
    @handshake = handshake.to_i
  end

  def commands
    EVENTS.filter.with_index do |_, i|
      (@handshake & 1 << i).nonzero?
    end.reverse_if((@handshake & REVERSE_EVENT).nonzero?)
  end
end
